controller/user-controllers.js

```diff
@@ -60,9 +60,16 @@ const logIn = async (req, res, next) => {
 const logOut = async (req, res, next) => {
   try {
     const user = await User.findById(req.user._id);
+    if (!user) {
```

Comment:
Nie trzeba tego sprawdzać. Zobacz, masz już req.user._id to znaczy że za pomocą middleware sprawdziłaś że uzytkownik ma już dostęp do tego miejsca.